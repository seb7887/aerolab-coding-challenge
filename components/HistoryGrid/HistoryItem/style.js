import styled from 'styled-components';

export const Container = styled.div`
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;
  font-size: 1rem;
  box-shadow: ${props => props.theme.productShadow};
  min-height: 250px;
  min-width: 200px;
  transition: all 0.3s;

  &:hover {
    transform: translateY(-1rem);
  }

  .description {
    display: flex;
    flex-direction: column;

    .name {
      font-weight: 600;
    }
  }
`;

export const Img = styled.img`
  width: 70%;
  height: 70%;
  object-fit: cover;
`;
