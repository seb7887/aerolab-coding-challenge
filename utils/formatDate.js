export default function formatDate(value) {
  const getNumber = number => (number < 10 ? `0${number}` : String(number));
  const date = new Date(value);

  return `${getNumber(date.getMonth() + 1)}/${getNumber(
    date.getDate()
  )}/${getNumber(date.getFullYear())} ${getNumber(
    date.getUTCHours()
  )}:${getNumber(date.getUTCMinutes())}`;
}
