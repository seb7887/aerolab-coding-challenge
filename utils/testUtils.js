import faker from 'faker';

export const fakeUser = {
  name: 'John Kite',
  points: '7000'
};

export const mockCategories = [
  'all',
  'gaming',
  'laptops',
  'tablets',
  'cameras',
  'audio',
  'tv',
  'drones',
  'accessories',
  'home'
];

export const fakeProduct = {
  _id: 7,
  name: 'iPhone',
  category: 'Phones',
  cost: 800,
  img: {
    url: '/static/image/image.png'
  }
};

export const generateProduct = () => {
  const newProduct = {
    _id: faker.random.uuid(),
    name: faker.commerce.productName(),
    category: faker.commerce.productAdjective(),
    cost: 800,
    img: {
      url: faker.image.imageUrl()
    }
  };
  return newProduct;
};

export const fakeProducts = [fakeProduct];

export const generateProducts = quantity => {
  let products = [];

  for (let i = 0; i < quantity; i++) {
    const newProduct = generateProduct();
    products.push(newProduct);
  }

  return products;
};
