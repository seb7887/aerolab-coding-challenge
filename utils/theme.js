const black = '#000';
const white = '#fff';
const cyan = '#0ad4fa';
const orange = '#ff6e05';
const red = 'red';
const green = 'green';

const bgLight = '#e9ebf1';
const bgDark = '#212531';
const secondaryBgLigth = 'rgba(0, 0, 0, 0.1)';
const secondaryBgDark = 'rgba(255, 255, 255, 0.1)';
const containerBgLight = '#fff';
const containerBgDark = '#000';
const boxShadowLight = '0px 0px 3px 0px rgba(0, 0, 0, 0.1)';
const boxShadowDark = 'rgba(31, 31, 31, 0.1)';
const productShadowLight = '1px 4px 5px 1px rgba(0, 0, 0, 0.1)';
const productShadowDark = '1px 4px 5px 1px rgba(255, 255, 255, 0.1)';

const primaryLight = '#616161';
const primaryDark = '#f8f8f8';

const secondaryLight = black;
const secondaryDark = white;
const greyLight = '#e0e0e0';
const greyDark = '#a3a3a3';

const theme = {
  light: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyLight,
    bg: bgLight,
    secondaryBg: secondaryBgLigth,
    containerBg: containerBgLight,
    boxShadow: boxShadowLight,
    productShadow: productShadowLight,
    primary: primaryLight,
    secondary: secondaryLight
  },
  dark: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyDark,
    bg: bgDark,
    secondaryBg: secondaryBgDark,
    containerBg: containerBgDark,
    boxShadow: boxShadowDark,
    productShadow: productShadowDark,
    primary: primaryDark,
    secondary: secondaryDark
  }
};

export default theme;
