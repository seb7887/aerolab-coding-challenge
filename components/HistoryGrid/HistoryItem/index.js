import React from 'react';
import PropTypes from 'prop-types';

import formatDate from '../../../utils/formatDate';

import { Container, Img } from './style';

const HistoryItem = ({ item }) => (
  <Container data-testid='history-item'>
    <div>
      <span data-testid='create-date'>{formatDate(item.createDate)}</span>
    </div>
    <Img src={item.img.url} alt={item.name} />
    <div className='description'>
      <span data-testid='category'>{item.category}</span>
      <span className='name' data-testid='name'>
        {item.name}
      </span>
    </div>
  </Container>
);

HistoryItem.propTypes = {
  item: PropTypes.object.isRequired
};

export default HistoryItem;
