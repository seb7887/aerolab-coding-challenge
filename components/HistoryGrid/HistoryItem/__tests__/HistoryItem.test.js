import { render } from 'react-testing-library';

import HistoryItem from '../index';

import { fakeHistoryItem } from '../../../../utils/testUtils';
import formatDate from '../../../../utils/formatDate';

describe('HistoryItem', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(<HistoryItem item={fakeHistoryItem} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display product description', () => {
    const { getByTestId } = render(<HistoryItem item={fakeHistoryItem} />);

    expect(getByTestId('create-date').textContent).toBe(
      formatDate(fakeHistoryItem.createDate)
    );
    expect(getByTestId('category').textContent).toBe(fakeHistoryItem.category);
    expect(getByTestId('name').textContent).toBe(fakeHistoryItem.name);
  });
});
