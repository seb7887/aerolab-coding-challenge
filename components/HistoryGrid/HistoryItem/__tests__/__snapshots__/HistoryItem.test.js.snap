// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="style__Container-sc-1ba92t9-0 dVxYRm"
    data-testid="history-item"
  >
    <div>
      <span
        data-testid="create-date"
      >
        04/24/2019 22:17
      </span>
    </div>
    <img
      alt="iPhone"
      class="style__Img-sc-1ba92t9-1 bIOfTM"
      src="/static/image/image.png"
    />
    <div
      class="description"
    >
      <span
        data-testid="category"
      >
        Phones
      </span>
      <span
        class="name"
        data-testid="name"
      >
        iPhone
      </span>
    </div>
  </div>
</DocumentFragment>
`;
